(#201) Desktop Header

Keegan-lee opened the pull request:
#### What's this PR do?
- Aligns Header items
- Adds Deposit and Withdraw buttons
- Conforms Header Items style to Zeplin designs

#### Where should the reviewer start?
Header.vue
HeaderUserInfo.vue

#### How should this be manually tested?
- Compare the working vue to the designs laid out on Zeplin

#### Any background context you want to provide?
#### What are the relevant tickets?
#### Screenshots (if appropriate)
#### URL
[http://staging-ui.trusty.apasia.tech:8080/branch-(-201)DesktopHeader](http://staging-ui.trusty.apasia.tech:8080/branch-(-201)DesktopHeader)
#### Questions:

Commit:
(#201) Desktop Header
- Add Deposit and Withdraw buttons to header
- Align Deopsit and Withdraw buttons to center
- Conform Deposit and Withdraw buttons to Zeplin design
- Add Account and down arrow to right side of header
- Made header same width as container

roma219 commented on src/views/Header/Header.vue:
```diff
@@ -1,14 +1,35 @@
+    <div class="lg:w-1/3">
+      <div class="lg:w-1/4 inline-block" />
+
+      <Button
```
Please use an existing `Button` component (`src/components/Button`) with type=`secondary` and size=`small`. If the size of the button is incorrect in this case, modify `Button` component `btn--small` styles.

Keegan-lee replied:
Thanks for the suggestions. I've made the changes. Looks much better now, I see in Zeplin where I need to pay attention to the styles. This should help.

roma219 commented on src/views/Header/HeaderUserInfo.vue:
```diff
@@ -26,7 +32,21 @@ export default {
+  p {
+    color: config('colors.grey');
+    width: 100%;
+    padding-right: 10px;
```
Please use `rem` instead of `px`

Keegan-lee replied:
Done.

roma219 reviewed (changes requested):
Needs to match the design:
![image](https://user-images.githubusercontent.com/4411295/49344424-0c785d00-f688-11e8-83d0-0fa39c691e64.png)

1) Change Bitshares Logo size
2) Header should have a background color, paddings & margin
3) Use existing `Button` component + modify it to match the design
4) Modify user name styling to match design (font size, weight, color)
5) Fix merge conflicts

Commit:
(#201)Desktop Header
- Fixed Button Style as on Zepplin
- Fixed Header Style, applied styles from Zeplin
- Fixed account name style, applied styles from Zeplin

Keegan-lee committed:
Merge branch 'dev' into (#201)DesktopHeader

roma219 commented on src/views/Header/Header.vue:
```diff
@@ -1,15 +1,39 @@
+        type="secondary"
+        size="small"
+      >Deposit</Button>
+      <Button
```
We have an existing UI components stored in `src/components`, please use `Button` component from there. Examine it & it's props and modify `small` appearance of the button to fit design.

roma219 commented on src/views/Header/Header.vue:
```diff
@@ -70,10 +94,43 @@ export default {
-  align-items: flex-end;
+  align-items: center;
+  height: 61px;
+  background-color: var(--black);
```
it should be black with opacity 0.2

roma219 commented on src/views/Header/Header.vue:
```diff
@@ -70,10 +94,43 @@ export default {
+  height: 61px;
+  background-color: var(--black);
+  margin: 5px 0px 5px 0px;
+  width: 1280px;
```
No need to set width hard like that - it results in this on smaller screens:
![image](https://user-images.githubusercontent.com/4411295/49423013-fa411080-f7a6-11e8-8187-4b792497f650.png)

Keegan-lee replied:
I found a way around this. This was tough, because of the header being black with opacity of .2. 

The contents of the header would have taken on the opacity (no way around this) of the background. 

So I set a backdrop to the background, and made that black with opacity .2. I made this the full width. Set position: absolute, and made it always stick to right: 0.

Take a look at the code. It looks good now.

roma219 replied:
> I found a way around this. This was tough, because of the header being black with opacity of .2. The contents of the header would have taken on the opacity (no way around this) of the background.

You can achieve it by setting background color to color of rgba format, e.g.:
```
background-color: rgba(0, 0, 0, 0.2)
```

roma219 reviewed (changes requested):
- > Header should have a background color, paddings & margin

Still no bg
![image](https://user-images.githubusercontent.com/4411295/49422317-7f76f600-f7a4-11e8-9a5f-5f14ede0dd8e.png)

- Needs to use Button component (`src/components/Button`)
- Please name branches like `123-desktop-header` (without `(`)

Commit:
#201 - Addressed PR Issues
- Made a background on the header
- Used Button from /components

Commit:
#201 - Removed Unused Style
- Remove style from buttons that is not being used.

roma219 commented on src/views/Header/Header.vue:
```diff
@@ -70,10 +96,47 @@ export default {
+  display: inline-block;
+}
+
+.header-btn {
```
Please replace this button styles with passing a prop `size="small"` to `Button`. After that, if it is needed, modify `Button` component to have appropriate style when prop `size="small"` is passed.

roma219 replied:
Was not resolved. You should modify Buttons sizes through classes, but instead modify `small` styles inside `Button` component

roma219 replied:
@Keegan-lee

roma219 commented on src/views/Header/HeaderUserInfo.vue:
```diff
@@ -26,7 +32,23 @@ export default {
+    width: 100%;
+    padding-right: 1rem;
+    font-weight: 500;
+    font-size: 15px;
```
We use `rem`s instead of `pixels`

roma219 replied:
Not resolved.

Change all occurances of `px` to `rem`s. 1rem = 16px.
https://medium.com/@julienetienne/pixels-are-dead-faa87cd8c8b9

roma219 replied:
@Keegan-lee

roma219 commented on src/views/Header/Header.vue:
```diff
@@ -1,27 +1,53 @@
+
+      <Button
+        text="Deposit"
+        class="header-btn lg:w-1/3"
```
Why does it have `lg:w-1/3` style?

roma219 commented on src/views/Header/Header.vue:
```diff
@@ -1,27 +1,53 @@
-      @clicked="handleDropdownClick"
-    />
+    <div class="back-header" />
+    <div class="lg:w-1/3">
```
These `<div class="lg:w-1/3">` are not needed here - they are gonna be displayed just fine as they are because of `display: flex` on `.header` container

Keegan-lee replied:
This class keeps the deposit/withdraw buttons centred on the page, the logo on the right, etc. The items will abide by the flex rules, but wont be consistently in place as shown on the designs.

roma219 reviewed (changes requested):
Better, but needs a bit of improvements

Commit:
#201
- Changed px to rem
- Removed unused style
- Remove unnecessary classes

roma219 commented on src/views/Header/Header.vue:
```diff
@@ -70,10 +96,43 @@ export default {
+  z-index: 0;
+}
+
+.inline-block {
```
No need for a separate class for that

roma219 reviewed (changes requested):
See prev discussions - some is not resolved.

Commit:
#201 Fixed Styles
- Moved button styles into the button component
- Changed px into rem
- Got rid of unneeded class and applied inline-block to parent
